pkg/cmd/repo/delete/delete.go

```diff
@@ -48,18 +48,21 @@ To authorize, run "gh auth refresh -s delete_repo"`,
 			if len(args) > 0 {
 				opts.RepoArg = args[0]
 			}
+
 			if !opts.IO.CanPrompt() && !opts.Confirmed {
-				return cmdutil.FlagErrorf("--confirm required when not running interactively")
+				return cmdutil.FlagErrorf("--yes or --confirm required when not running interactively")
 			}
 
 			if runF != nil {
 				return runF(opts)
 			}
+
 			return deleteRun(opts)
 		},
 	}
 
-	cmd.Flags().BoolVar(&opts.Confirmed, "confirm", false, "confirm deletion without prompting")
+	cmd.Flags().BoolVar(&opts.Confirmed, "confirm", false, "[deprecated] confirm deletion without prompting")
```

Comment:
Instead of adding `[deprecated]` to flag description, you can add this:
```go
cmd.Flags().MarkDeprecated("confirm", "explanation goes here")
```
That way a warning will be printed for those who still use the deprecated flag.